cgi/import_photos_upload.pl

```diff
@@ -64,153 +66,24 @@
-	tagContent.attr("name", "tag_" + field_value_number);
-	tagContent.attr("id", "tag_" + field_value_number);
-	tagContent.val("");
-
-	return element;
-}
-
-function addAddFieldValue(target, field_values_number) {
-	var addFieldValue1 = modifyAddFieldValue(\$(".add_field_values_row").first().clone(), field_values_number);
-
-	\$(".add_field_values_row").last().after(addFieldValue1);
-}
-</script>
-JS
-;
-
-	$initjs .= <<JS
-
-	//On tag field value change
-	\$(document).on("change", ".tag-add-field-value > input", function(e){
-		var field_value_number = parseInt(e.target.name.substr(e.target.name.length - 1));
-		//If it's the last field value, add one more
-		if(!isNaN(field_value_number) && \$("#tag_" + (field_value_number + 1).toString()).length === 0){
-			addAddFieldValue(e.target, field_value_number + 1);
-		}
-	});
-	\$(document).on("change", ".tag-add-field > select", function(e){
-		var field_value_number = parseInt(e.target.name.substr(e.target.name.length - 1));
-		//If it's the last field value, add one more
-		if(!isNaN(field_value_number) && \$("#tag_" + (field_value_number + 1).toString()).length === 0){
-			addAddFieldValue(e.target, field_value_number + 1);
-		}
-	});
-JS
-;
-
-	# File upload button
-
-	$html .= <<HTML
-        <!-- The fileupload-buttonbar contains buttons to add/delete files and start/cancel the upload -->
-        <div class="row fileupload-buttonbar">
-          <div class="large-7 columns">
-            <!-- The fileinput-button span is used to style the file input field as button -->
-            <span class="button small btn-success fileinput-button">
-              @{[ display_icon('add') ]}
-              <span>$Lang{add_photos}{$lang}</span>
-              <input type="file" name="files[]" multiple accept="image/*" data-url="/cgi/product_image_import.pl" />
-            </span>
-            <!-- The global file processing state -->
-            <span class="fileupload-process"></span>
-          </div>
-          <!-- The global progress state -->
-          <div class="large-5 columns fileupload-progress fade">
-            <!-- The global progress bar -->
-            <div
-              class="progress progress-striped active"
-              role="progressbar"
-              aria-valuemin="0"
-              aria-valuemax="100"
-            >
-              <div
-                class="progress-bar progress-bar-success meter"
-                style="width:0%;"
-              ></div>
-            </div>
-            <!-- The extended global progress state -->
-            <div class="progress-extended">&nbsp;</div>
-          </div>
-        </div>
-        <!-- The table listing the files available for upload/download -->
-        <table role="presentation" class="table table-striped">
-          <tbody class="files"></tbody>
-        </table>
-      </form>
-
-	<div id="empty_space_for_equalizer" style="height:200px;width:100px;">&nbsp;</div>
-
-HTML
-;
+  my $i = 0;
+  $template_data_ref->{i} = $i;
```

Comment:
$i is always 0 in fact, so you can remove it, and replace all instances of [% i %] by 0.

Reply:
yeah, I checked that, `$i` is 0 everywhere, but it thought to keep it generic, in case of modifications. If someone modifies the code, he/she will only have to change the value of `$i` instead of replacing all the variables.

Reply:
ok